clients/venice-producer/src/main/java/com/linkedin/venice/producer/online/ProducerTool.java

```diff
@@ -193,17 +196,23 @@ private static void writeToStore(ProducerContext producerContext) throws Excepti
 
   private static Object getValueObject(String valueString, RouterBasedStoreSchemaFetcher schemaFetcher) {
     Object value = null;
+    List<Exception> exceptionList = new ArrayList<>();
     for (Schema valueSchema: schemaFetcher.getAllValueSchemas()) {
       try {
         value = adaptDataToSchema(valueString, valueSchema);
         break;
       } catch (Exception e) {
-        // Nothing to do. Try the next schema
+        exceptionList.add(e);
+        // Try the next schema
       }
     }
     if (value == null) {
+      String exceptionDelimiter = "\n\t";
+      String exceptionDetails =
+          exceptionList.stream().map(Throwable::getMessage).collect(Collectors.joining(exceptionDelimiter));
       throw new VeniceException(
-          "Value schema not found. Is a schema that is compatible with the specified data already registered?");
+          "Value schema not found. Is a schema that is compatible with the specified data already registered?\nException messages for each schema (not in order):"
```

Comment:
possible to add schema id to corresponding stack trace ? (probably will need a different API to fetch)

Reply:
Yeah. We don't have that info currently. We'll need to add a new API